### PR TITLE
Fixed Tileset Editor bug related to polygon size

### LINKED ISSUE #27035: Tileset editor crashes when editing 3.0.6 made resource
**Godot version:**

Godot 3.1 stable

**OS/device including version:**

Windows 10 x64

**Issue description:**

Could be related to #21545

I've opened a project that I've been working on since 3.0.6 in 3.1, and wanted to try out the Tileset editor. Once I opened the editor, I tried to edit the Tileset, and it crashed. It worked for a polygon collision, but when editing a tile with rectangle polygon collision, it crashed.

**Steps to reproduce:**

For the minimal reproduction project:

1. Open project with Godot 3.1 stable
1. Double-click mocktiles.tres to open the Tileset editor
1. Single-click "window"
1. Double-click sprite to edit collision -> crash

Sprites with rectangular collision crash, but polygons seem to work.

**Minimal reproduction project:**

[TilesetTest.zip](https://github.com/godotengine/godot/files/2964511/TilesetTest.zip)

### LINKED ISSUE #27063: CollisionObject signals do not trigger
**Godot version:**
3.1

**Issue description:**
(3D)

When I changed to Godot 3.1 and opened my project from version 3.0.6 I realized that mouse_entered 
signals were not triggering, but they worked before.
So I was asking in 2 Discord servers without getting any answer, tried then to do a small project and got the exact same results.

The signals for area2D's work tho.

**Steps to reproduce:**
(3D)

1) Create new project
2) Add Area with Collision Shape
3) Connect mouse_entered signal with any script
4) Try to get mouse_entered signal triggered

[CollisionObject Signals Debug.zip](https://github.com/godotengine/godot/files/2968212/CollisionObject.Signals.Debug.zip)

### PR DESCRIPTION
**Pull request by siddharth952**
Fixes #27035

1)
Crash was due to null case not taken into account for polygon.size, added if case to fix the problem of not checking for polygon.size() == 0

gitissue

### CODE REVIEW

**Comment by siddharth952** [comment]
@Chaosus Any ideas why Travis CI build is failing? build is successful on my local machine.

**Comment by Chaosus** [comment]
I think you should use clang-format tool (best way to do it using VSCode extension) - see https://docs.godotengine.org/en/latest/community/contributing/code_style_guidelines.html

**Comment by siddharth952** [comment]
@Chaosus Thank you.

**Comment by Chaosus** [comment]
You welcome. You should also squash the commits together as its also required to our pipeline - see https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase

File: editor/plugins/tile_set_editor_plugin.cpp [hunk 1]
```diff
@@ -2417,7 +2417,9 @@ void TileSetEditor::draw_polygon_shapes() {
 							colors.push_back(c_bg);
 						}
 					}
-					if (polygon.size() > 2) {
+					if (polygon.size() == 0)
+						continue;
+					else if (polygon.size() > 2) {
```
Comment (YeldhamDev) [inline editor/plugins/tile_set_editor_plugin.cpp]: This `else` is not necessary, as the `continue` just skips to the next iteration.

Comment (siddharth952) [reply editor/plugins/tile_set_editor_plugin.cpp]: Yes, thanks for pointing that out!

**Comment by akien-mga** [comment]
#27063 has been fixed by #27200, please rebase to keep only the TileSet Editor changes.

For future PRs, please do one PR for each unrelated change to avoid such issue.

**Comment by akien-mga** [comment]
Still needs to be rebased. Please make sure that the resulting commit log is clear.
See http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html

**Comment by siddharth952** [comment]
@akien-mga Yes I did try, there seems to be a problem with the branch I'll sort it out in 1-2 days

**Comment by YeldhamDev** [comment]
I completely forgot this PR, I superseded this with mine: #27602

**Review by MarianoGnu — APPROVED** [review]

**Comment by akien-mga** [comment]
@MarianoGnu Please don't merge a PR when it has a messy commit history. This PR has 4 commits for a +2 change, that doesn't fit our contribution guidelines.

**Comment by akien-mga** [comment]
I even asked twice for a rebase above, please read pending comments before merging.

**Comment by akien-mga** [comment]
We can skip cherry-picking this 4 commits anyway, they're superseded by #27602.

**Pull request merged**